docs: workaround for kube-oidc-proxy 2.1.1 (#4054)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.1/release-notes/2.1.1/index.md

- traefik-forward-auth: 0.3.2
- velero: 3.1.3

## Known Issues 

> 

Reviewer: ```suggestion

The following items are known issues with this release.

```

### Create cert-manager resources on attached clusters with cert-manager pre-installed

> If you are attaching a cluster that already has `cert-manager` installed, you need to manually create some cert-manager resources after attaching your cluster.

Reviewer: ```suggestion
If you attach a cluster that already has `cert-manager` installed, you need to manually create the three cert-manager resources, in the yaml file, after attaching your cluster.
```

For example, [Konvoy-created clusters that are self-managed][konvoy-self-managed] have `cert-manager` already installed to the `cert-manager` namespace.

> Check whether your cluster is self-managed by:

Reviewer: ```suggestion
Verify that your cluster is self-managed by:
```

```bash
export KUBECONFIG=<kubeconfig-path>
export WORKSPACE_NAMESPACE=<your-workspace-namespace>

> kubectl get kommandercluster -n $WORKSPACE_NAMESPACE

Author: Want to double check if this method can be described like this.

Reviewer 2: Do you need some assist testing this?

Reviewer 3: `kommandercluster` is a kommander resource that only exists on the management cluster

Author: gonna change it to check cert-manager pod instead as discussed above

```

> If your cluster is not self-managed, it produces output like:

Reviewer: ```suggestion
If your cluster is not self-managed, the output from the previous command will resemble this example:
```

```bash
error: the server doesn't have a resource type "kommandercluster"
```

> Otherwise, it should produce output like:

Reviewer: ```suggestion
If your cluster is self-managed, then the output resembles this example:
```

```bash
NAME     AGE   PHASE
my-cluster   30m   Provisioned
```

> Create the following yaml file only if your cluster is self-managed:

Reviewer: ```suggestion
If your cluster is self-managed, then create the following yaml file:
```

```yaml
cat << EOF > cert_manager_root-ca.yaml
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: kommander-bootstrap-ca-issuer
  namespace: $WORKSPACE_NAMESPACE
spec:
  selfSigned: {}
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: kommander-bootstrap-root-certificate
  namespace: $WORKSPACE_NAMESPACE
spec:
  commonName: ca.kommander-bootstrap
  dnsNames:
    - ca.kommander-bootstrap
  duration: 8760h
  isCA: true
  issuerRef:
    name: kommander-bootstrap-ca-issuer
  secretName: kommander-bootstrap-root-ca
  subject:
    organizations:
      - cert-manager
---
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: kommander-bootstrap-issuer
  namespace: $WORKSPACE_NAMESPACE
spec:
  ca:
    secretName: kommander-bootstrap-root-ca
EOF
```

> Then, apply this file to your cluster:

Reviewer: ```suggestion
Next, apply this file to your cluster:
```

```sh
kubectl apply -f cert_manager_root-ca.yaml
```

> Now, fix the broken certificates for the attached cluster:

Reviewer: ```suggestion
Finally, fix the broken certificates for the attached cluster:
```

- traefik-forward-auth: 0.3.2
- velero: 3.1.3

> ## Known Issues 

Reviewer 2: linting

```suggestion
## Known Issues
```

The following items are known issues with this release.

### Create cert-manager resources on attached clusters with cert-manager pre-installed

If you attach a cluster that already has `cert-manager` installed, you need to manually create the three cert-manager resources, in the yaml file, after attaching your cluster.

For example, [Konvoy-created clusters that are self-managed][konvoy-self-managed] have `cert-manager` already installed to the `cert-manager` namespace.

> Verify that your cluster is self-managed by:

Reviewer 2: This has the possibility to exist outside of DKP created clusters probably, right? Most likely, if they're using DKP/Kommander 2.1.1, it's going to be a Konvoy created cluster, so this would work.

```suggestion
Verify that your DKP cluster is self-managed by:
```

Reviewer 3: do we actually need to check if a cluster is self-managed, or do we just need to check if cert-manager is already installed?

```bash
export KUBECONFIG=<kubeconfig-path>
export WORKSPACE_NAMESPACE=<your-workspace-namespace>
kubectl get kommandercluster -n $WORKSPACE_NAMESPACE
```

If your cluster is not self-managed, the output resembles this example:

```bash
error: the server doesn't have a resource type "kommandercluster"
```

If your cluster is self-managed, the output resembles this example:

```bash

> NAME     AGE   PHASE
> my-cluster   30m   Provisioned

Reviewer 2: ```suggestion
NAME         AGE   PHASE
my-cluster   30m   Provisioned
```

```

> If your cluster is self-managed, then create the following yaml file:

Reviewer 4: what should happen if the cluster is not self-managed?

Author: @Colin1841 changed it to this way and original one was "Create the following yaml file only if your cluster is self-managed:"

If not self-managed, you don't need to apply this yaml

```yaml
cat << EOF > cert_manager_root-ca.yaml
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: kommander-bootstrap-ca-issuer
  namespace: $WORKSPACE_NAMESPACE
spec:
  selfSigned: {}
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: kommander-bootstrap-root-certificate
  namespace: $WORKSPACE_NAMESPACE
spec:
  commonName: ca.kommander-bootstrap
  dnsNames:
    - ca.kommander-bootstrap
  duration: 8760h
  isCA: true
  issuerRef:
    name: kommander-bootstrap-ca-issuer
  secretName: kommander-bootstrap-root-ca
  subject:
    organizations:
      - cert-manager
---
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: kommander-bootstrap-issuer
  namespace: $WORKSPACE_NAMESPACE
spec:
  ca:
    secretName: kommander-bootstrap-root-ca
EOF
```

> Next, apply this file to your cluster:

Reviewer 2: nit: just some clarifying language.

```suggestion
Next, apply this file to your self-managed cluster you are attaching to Kommander:
```

Reviewer 2: I think this is correct, right?

```sh
kubectl apply -f cert_manager_root-ca.yaml
```

> Finally, fix the broken certificates for the attached cluster:

Reviewer 4: does this need to be run on all attached clusters?

Author: this is the common step for both self-managed / non self-managed attached clusters. and yes, needs to be run on all attached clusters.

Author: note instead of checking "self-managed", we gonna only care about whether cert-manager is pre installed

```bash
kubectl patch certificate -n $WORKSPACE_NAMESPACE kube-oidc-proxy --type='merge' -p '{"spec": {"issuerRef": {"kind": "Issuer", "name": "kommander-bootstrap-issuer"}}}'
kubectl patch certificate -n $WORKSPACE_NAMESPACE kommander-traefik --type='merge' -p '{"spec": {"issuerRef": {"kind": "Issuer", "name": "kommander-bootstrap-issuer"}}}'
```

> Cert-manager will fail to deploy due to your existing `cert-manager` installation. This is expected and can be ignored.

Reviewer 2: Totally unnecessary here, but thought this should be consistent.

```suggestion
`cert-manager` will fail to deploy due to your existing `cert-manager` installation. This is expected and can be ignored.
```
